LXReorderableCollectionViewFlowLayout/LXReorderableCollectionViewFlowLayout.m

```diff
@@ -138,7 +140,10 @@ - (void)applyLayoutAttributes:(UICollectionViewLayoutAttributes *)layoutAttribut
 }
 
 - (id<LXReorderableCollectionViewDataSource>)dataSource {
-    return (id<LXReorderableCollectionViewDataSource>)self.collectionView.dataSource;
+    if ( _dataSource == nil ) {
+        _dataSource = (id<LXReorderableCollectionViewDataSource>)self.collectionView.dataSource;
+    }
+    return _dataSource;
```

Comment:
I'd have the property declared read only and just return `self.collectionView.dataSource`.   If the collection view changed the data source, this would not be very happy.